### PR TITLE
Add perspective projection matrix support

### PR DESCRIPTION
**Pull request by IMS212**
Simple code.

I have not tested this on an actual pack yet, however the matrix completely matches OptiFine's as well as Mojang's "perspective" function.

### CODE REVIEW

**Comment by IMS212** [comment]
Thanks- wait a second

Though seriously, this is pretty simple, would someone be able to make a test pack?

**Comment by coderbot16** [comment]
Honestly I'm not sure if any packs made in the past decade actually use FOV in the shadow pass. Seems likely that it was just legacy silliness.

File: src/main/java/net/coderbot/iris/shadow/ShadowMatrices.java [hunk 1]
```diff
@@ -23,6 +23,21 @@
 		};
 	}
 
+	public static float[] createPerspectiveMatrix(float fov, float aspect) {
+		float i = (float)(1.0D / Math.tan(fov * 0.01745329238474369D / 2.0D));
+		Matrix4f matrix4f = new Matrix4f();
+		return new float[] {
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/shadow/ShadowMatrices.java:29]: This code gives me the feeling that it was copied from somewhere, with the arbitrary constants and the unused Matrix4f. I'd prefer if this code weren't copied & had more documentation, such as what units FOV has.

File: src/main/java/net/coderbot/iris/pipeline/ShadowRenderer.java [hunk 2]
```diff
@@ -75,8 +72,9 @@
 	private final float renderDistanceMultiplier;
 	private final int resolution;
 	private final float intervalSize;
+	private final Float fov;
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/pipeline/ShadowRenderer.java:75]: What units is FOV in? (degrees or radians) - this is important to document in all the places we have an FOV field.

**Review by coderbot16 — CHANGES_REQUESTED** [review]

File: src/main/java/net/coderbot/iris/shadow/ShadowMatrices.java (hunk 1 above)
Comment (IMS212) [reply src/main/java/net/coderbot/iris/shadow/ShadowMatrices.java:29]: Correct, it was copied from Vanilla's projection method. I'll look into this.

**Comment by coderbot16** [comment]
https://github.com/IrisShaders/Iris/issues/893

**Comment by IMS212** [comment]
Fixed.

File: src/main/java/net/coderbot/iris/shadow/ShadowMatrices.java [hunk 3]
```diff
@@ -23,6 +23,21 @@
 		};
 	}
 
+	public static float[] createPerspectiveMatrix(float fov) {
+		// This converts from degrees to radians.
+		float rad = (float) (1.0f / Math.tan(Math.toRadians(fov) * 0.5f));
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/shadow/ShadowMatrices.java:28]: This concerts to radians, but just to call tan(). A more accurate name is yScale.

**Review by coderbot16 — CHANGES_REQUESTED** [review]

File: src/main/java/net/coderbot/iris/shadow/ShadowMatrices.java (hunk 3 above)
Comment (IMS212) [reply src/main/java/net/coderbot/iris/shadow/ShadowMatrices.java]: Done

**Review by coderbot16 — APPROVED** [review]
Thanks!

**Pull request merged**